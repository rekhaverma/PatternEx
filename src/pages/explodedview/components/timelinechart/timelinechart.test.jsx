import React from 'react';
import moment from 'moment';
import TimelineChart from './timelinechart.component';

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<TimelineChart />', () => {
  it('should match with snapshot', () => {
    const props = {
      data: [
        {
          color: '#ff0000',
          create_time: 'Sat, 23 Jun 2018 09:03:00 -0000',
          description: 'Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information',
          id: '98ca552c-bff2-4969-9d18-bcc4da4759f1',
          name: ['CC', 'Fraud'],
          severity: 4,
          type: 'M',
        }],
      startDate: moment.utc('2018-01-30').subtract(30, 'days').startOf('day'),
      endDate: moment.utc('2018-01-30').startOf('day'),
    };
    const wrapper = mount(<TimelineChart {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
