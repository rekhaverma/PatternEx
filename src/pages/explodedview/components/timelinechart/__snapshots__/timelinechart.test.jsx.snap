// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineChart /> should match with snapshot 1`] = `
<TimelineChart
  data={
    Array [
      Object {
        "color": "#ff0000",
        "create_time": "Sat, 23 Jun 2018 09:03:00 -0000",
        "description": "Credit Card Fraud. Account created using compromised/stolen credit card details and personal identity information",
        "id": "98ca552c-bff2-4969-9d18-bcc4da4759f1",
        "name": Array [
          "CC",
          "Fraud",
        ],
        "severity": 4,
        "type": "M",
      },
    ]
  }
  endDate={"2018-01-30T23:59:59.999Z"}
  startDate={"2017-12-31T00:00:00.000Z"}
>
  <div
    className="svgContainer"
  >
    <svg
      className="timeline__chart"
      height={NaN}
      width={NaN}
    >
      <g
        className="rectangle"
      />
      <g
        className="circles"
      />
      <g
        className="labels"
      />
      <g
        className="timestamps"
      />
      <defs
        className="gradient"
      />
    </svg>
  </div>
</TimelineChart>
`;
